tribits/ci_support/CDashQueryAnalyzeReport.py

```diff
@@ -2193,6 +2410,415 @@ def createCDashTestHtmlTableStr(
+  #   size will be written on one line ending with ``<br>\n``.
+  #
+  #   cdashReportData.htmlEmailBodyBottom: Summary HTML table (with title)
+  #   will be written, along with formatting.
+  #
+  def reportSingleTestset(self, testsetTypeInfo, testsetTotalSize, testsetLOD,
+      sortTests=True,
+      limitTableRows=None,   # Change to 'int' > 0 to limit table rows
+      getTestHistory=False,
+    ):
+
+    testsetSummaryStr = \
+      getCDashDataSummaryHtmlTableTitleStr(testsetTypeInfo.testsetDescr,
+        testsetTypeInfo.testsetAcro, testsetTotalSize)
+
+    if self.verbose:
+      print("")
+      print(testsetSummaryStr)
+
+    if testsetTotalSize > 0:
+
+      if testsetTypeInfo.existanceTriggersGlobalFail:
+        self.cdashReportData.globalPass = False
+
+      self.cdashReportData.summaryLineDataNumbersList.append(
+        testsetTypeInfo.testsetAcro+"="+str(testsetTotalSize))
+
+      self.cdashReportData.htmlEmailBodyTop += \
+        colorHtmlText(testsetSummaryStr, testsetTypeInfo.testsetColor)+"<br>\n"
+
+      if sortTests or limitTableRows:
+        testsetSortedLimitedLOD = sortAndLimitListOfDicts(
+          testsetLOD, self.testDictsSortKeyList, limitTableRows )
+      else:
+        testsetSortedLimitedLOD = testsetLOD
+
+      if getTestHistory and self.addTestHistoryStrategy:
+        self.addTestHistoryStrategy.getTestHistory(testsetSortedLimitedLOD)
+
+      self.cdashReportData.htmlEmailBodyBottom += createCDashTestHtmlTableStr(
+        testsetTypeInfo, testsetTotalSize, testsetSortedLimitedLOD,
+        limitRowsToDisplay=limitTableRows,
+        htmlStyle=self.htmlStyle, htmlTableStyle=self.htmlTableStyle )
+
+
+# Class to generate the data for an HTML report for all test-sets represented
+# in a list of test dicts.
+# 
+class TestsetsReporter(object):
+
+  # Constructor
+  #
+  # cdashReportData [persisting]: Data used to create the final report (of type
+  # CDashReportData).
+  #
+  def __init__(self, cdashReportData, testsetAcroList=getStandardTestsetAcroList(),
+      htmlStyle=None, htmlTableStyle=None, verbose=True,
+    ):
+    self.cdashReportData = cdashReportData
+    self.testsetAcroList = testsetAcroList
+    self.singleTestsetReporter = SingleTestsetReporter(cdashReportData,
+      htmlStyle=htmlStyle, htmlTableStyle=htmlTableStyle, verbose=verbose)
+
+
+  # Separate out and report on all of the test-sets in the input list of test
+  # dicts.
+  # 
+  def reportTestsets(self, testsLOD):
+    testDictsByTestsetAcro = binTestDictsByTestsetAcro(testsLOD)
+    for testsetAcro in self.testsetAcroList:
+      testsetLOD = testDictsByTestsetAcro.get(testsetAcro, None)
+      if testsetLOD:
+        testsetTypeInfo = getStandardTestsetTypeInfo(testsetAcro)
+        self.singleTestsetReporter.reportSingleTestset(testsetTypeInfo,
+          len(testsetLOD), testsetLOD)
+  # ToDo: Modify the above to assert that there are no unexpected test-set
+  # types!
```

Comment:
I need to add a test for this and do this assert.  Otherwise, we are asking for trouble later when we add new types of test-sets (e.g; 'twiatff', 'twimb', etc.)

Reply:
I will just take out the sublist.